src/webview/MessageList.js

```diff
@@ -309,8 +285,16 @@ class MessageList extends Component<Props> {
     // https://github.com/react-native-community/react-native-webview/pull/697
     return (
       <WebView
-        source={{ baseUrl, html }}
-        originWhitelist={['file://']}
+        source={{ baseUrl: baseUrl.toString(), html }}
+        originWhitelist={[
+          // `baseUrl.origin` may be "null" (an "opaque origin") for
+          // `file://` URLs. [1] [2] `react-native-webview` doesn't
+          // know about this.
+          //
+          // [1] https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Resource.20URLs/near/926650
+          // [2] https://github.com/jsdom/whatwg-url/tree/v7.0.0/#specification-conformance
+          baseUrl.origin === new WhatwgURL('file:///').origin ? 'file://' : baseUrl.origin,
```

Comment:
Oh, right—in this revision, I've acted on what I said [here](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Resource.20URLs/near/926696) and abandoned ever pointing to what's served by the packager.

In effect, this means the URL should always start with `file://`—so we could just say "file://", if that's preferred.

Reply:
I was just about to ask what this code was doing and why this part became more complicated :slightly_smiling_face: I guess that's the answer!

Yeah, let's leave this in the simple form it's in until/unless we find a need to do something more complex.